### DISCUSSION TITLE
Scaling an HPA's minReplicas value dynamically

### DISCUSSION THREAD

**Discussion by sthomson-wyn** [kedacore/keda#5512] [Q&A / Need Help]
Scaling a resource's replica count is simple enough, as long as there is a scale subresource pointing to the replica count

But what if we want to scale an HPA's (or any autoscaling controller's) minReplicas (or maxReplicas) value? Is this possible with keda?

Our use-case is changing a controller's minReplicas value to 0 during a set time period, and > 0 during another set time period. This could be done easily with a cron job / patching, but I'm curious if such funcionality exists in keda, or if it's maybe something that keda should handle

**Comment by zroubalik** [comment]
At the moment this is not covered by KEDA

**Comment by zerodayyy** [comment]
@sthomson-wyn you can set actual minReplicas to the smallest number you expect, and add cron triggers with relevant target replicas for respective time intervals, which would essentially act as minReplicas for that period; since by default HPA chooses the largest of desired replicas numbers across multiple metrics provided — it will not drop below whatever number you provide in cron trigger.
